homeassistant/components/snapcast/__init__.py

```diff
@@ -1 +1,49 @@
 """The snapcast component."""
+
+import asyncio
+import voluptuous as vol
+
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+DOMAIN = 'snapcast'
+
+SERVICE_SNAPSHOT = 'snapshot'
+SERVICE_RESTORE = 'restore'
+SERVICE_JOIN = 'join'
+SERVICE_UNJOIN = 'unjoin'
+
+ATTR_MASTER = 'master'
+
+SERVICE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+JOIN_SERVICE_SCHEMA = SERVICE_SCHEMA.extend({
+    vol.Required(ATTR_MASTER): cv.entity_id,
+})
+
+DATA_SERVICE_EVENT = 'snapcast_service_idle'
+
+
+async def async_setup(hass, config):
+    """Handle service configuration."""
+    hass.data[DATA_SERVICE_EVENT] = asyncio.Event()
+
+    async def service_handle(service):
+        """Dispatch a service call."""
+        hass.data[DATA_SERVICE_EVENT].clear()
+        async_dispatcher_send(hass, DOMAIN, service.service, service.data)
+        await hass.data[DATA_SERVICE_EVENT].wait()
```

Comment:
Why not pass the event via the dispatcher? that way it will be possible to run multiple services at the same time.

Reply:
Honestly, I don't really know I did how it was in the sonos component as I'm starting with asyncio.